tools/mpy-tool.py

```diff
@@ -1489,14 +1489,13 @@ def freeze_mpy(firmware_qstr_idents, compiled_modules):
     print("    true, // is_sorted")
     print("    %u, // allocated entries" % qstr_pool_alloc)
     print("    %u, // used entries" % len(new))
-    print("    (qstr_hash_t *)mp_qstr_frozen_const_hashes,")
+    if config.MICROPY_QSTR_BYTES_IN_HASH:
+        print("    (qstr_hash_t *)mp_qstr_frozen_const_hashes,")
     print("    (qstr_len_t *)mp_qstr_frozen_const_lengths,")
     print("    {")
     for _, _, qstr, qbytes in new:
         print('        "%s",' % qstrutil.escape_bytes(qstr, qbytes))
-        qstr_content += (
-            config.MICROPY_QSTR_BYTES_IN_LEN + config.MICROPY_QSTR_BYTES_IN_HASH + len(qbytes) + 1
-        )
+    qstr_content = qstr_size["metadata"] + qstr_size["data"] + 1
```

Comment:
The `+ 1` should probably be `+ len(new)`, because it's counting the terminating null byte on each string data.

Reply:
Reworked this to make it clearer.